src/lib.rs

```diff
@@ -597,6 +597,190 @@ impl<A: Array> ArrayVec<A> {
     }
 }
 
+impl<A: Array> ArrayVec<A>
+    where A::Item: Clone,
+{
```

Comment:
Style wise I'd prefer a single `impl` block. The where clause can go on each method fine. Split `impl` blocks feels archaic.